### ISSUE TITLE
CLI Colocation status for VMs and PPG is shown as "null"

### ISSUE THREAD

**Issue by ashutoshtandon** [Azure/azure-cli#24638]
Command:
az ppg show --name <name of ppg> --resource-group <name of resource group>

In portal, correct colocation status is shown but CLI output shows both PPG and VMs colocation status as 'null'

Related bug: 
https://dev.azure.com/msazure/One/_workitems/edit/16163797/

**Comment by ashutoshtandon** [comment]
![Screenshot (1)](https://user-images.githubusercontent.com/118155186/201584300-e4224db9-a4b9-4b78-a25a-2549ad3f06d0.png)
![Screenshot (5)](https://user-images.githubusercontent.com/118155186/201584310-0184afa8-13ba-47a3-ae07-315be2ce2f8c.png)
![Screenshot (4)](https://user-images.githubusercontent.com/118155186/201584314-2bf1f464-25e8-4c04-a07b-e8be4255e561.png)
![Screenshot (3)](https://user-images.githubusercontent.com/118155186/201584318-79e64e98-26bb-4352-be29-5c53919c733f.png)
![Screenshot (2)](https://user-images.githubusercontent.com/118155186/201584321-94d0a5a4-92dd-48ed-93f7-171018a8f695.png)

**Comment by yonzhan** [comment]
@zhoxing-ms for awareness

**Comment by zhoxing-ms** [comment]
@ashutoshtandon Could you please add `--include-colocation-status` to have a try?

**Comment by ashutoshtandon** [comment]
Seems to be working.

**Comment by ashutoshtandon** [comment]
The issue is reproducible when all VMs and PPG are in 'Aligned' state. In this scenario, VM Colocation status is coming as Null.
Although through PowerShell, **Get-AzVM** cmdlet is giving right output for VM colocation status.

**Comment by ashutoshtandon** [comment]
![Screenshot (11)](https://user-images.githubusercontent.com/118155186/201679873-49225cab-796b-45d6-8ca0-cdd012c2e2aa.png)
![Screenshot (10)](https://user-images.githubusercontent.com/118155186/201679891-46bf86a8-f436-45ea-ae07-6b4577ea61d5.png)

**Comment by ashutoshtandon** [comment]
![Screenshot (10)](https://user-images.githubusercontent.com/118155186/201681164-756d2b87-d414-4d3e-9882-b4b57e3ea370.png)
![Screenshot (11)](https://user-images.githubusercontent.com/118155186/201681180-122f1325-d84f-415e-ab2d-0a05d1954079.png)

**Comment by ashutoshtandon** [comment]
![Screenshot (14)](https://user-images.githubusercontent.com/118155186/201684734-3ed94851-ad7d-454c-bacf-01a3991d3e32.png)
![Screenshot (15)](https://user-images.githubusercontent.com/118155186/201684749-6fd29077-bcdd-4a9b-b4cd-ce0d3efe1f5c.png)
![Screenshot (12)](https://user-images.githubusercontent.com/118155186/201684757-18c6d84e-2edf-47ab-991e-d2825225e772.png)
![Screenshot (13)](https://user-images.githubusercontent.com/118155186/201684767-abbdf604-c9a8-43f8-b224-cb0c95e70549.png)

**Comment by ShivapriyaGujjarlapudi** [comment]
@zhoxing-ms - Could you please help reopen this for triage

**Comment by zhoxing-ms** [comment]
> VM Colocation status is coming as Null

@ShivapriyaGujjarlapudi This should be a service issue. The REST [proximity-placement-groups/get](https://learn.microsoft.com/en-us/rest/api/compute/proximity-placement-groups/get?tabs=HTTP) return the colocation status of VM as `null`.

**Comment by zhoxing-ms** [comment]
@ShivapriyaGujjarlapudi I'd like to confirm with you that all cases where colocation status of VM is `null` can be regarded as `Aligned`? Is there really a case of `null`?
In addition, it seems inappropriate for PowerShell to modify the return colocation status from REST service, which will lead to inconsistent experience between multiple client tools and REST. @dcaro Could you please help to confirm the reason for this design, and do you suggest that CLI also follow this design?

**Comment by ashutoshtandon** [comment]
@zhoxing-ms 
As per current implementation. REST [proximity-placement-groups/get](https://learn.microsoft.com/en-us/rest/api/compute/proximity-placement-groups/get?tabs=HTTP) returns colocat
![Screenshot (16)](https://user-images.githubusercontent.com/118155186/202422595-28bb08ac-b349-43e5-b64f-b4fe829a89c0.png)
ion status of VMs as `null` when overlying PPG is Aligned.
In Powershell, `Get-AzProximityPlacementGroup` cmdlet doesn't return colocation status of VMs. 
To get colocation status of VMs, `Get-AzVM` cmdlet is used which is working fine, returning `Aligned `status.

**Comment by zhoxing-ms** [comment]
@ashutoshtandon Yes, the `az vm show` command should be able to output the `Aligned` for colocation status of VMs. 
May I ask whether the REST [proximity-placement-groups/get](https://learn.microsoft.com/en-us/rest/api/compute/proximity-placement-groups/get?tabs=HTTP) can modify the return colocation status as `Aligned` to make the experience of multiple client tools and REST consistent?

**Comment by ashutoshtandon** [comment]
So, this is a GAed feature, and all the consumers of this API have translated its output as per their usage. Hence, making a change to this API would require:
1. Strong business reason for doing it.
2. Regression to prevent failures on consumer end

It was a PM ask to avoid redundant data at VM level for 'Aligned' PPGs which implicitly mean all underlying VMs are aligned. It was making the API response bloated.

**Comment by ShivapriyaGujjarlapudi** [comment]
@zhoxing-ms - Considering @ashutoshtandon 's inputs,could we make the necessary changes to the CLI output to keep it consistent with the GA-ed CRP response

**Comment by zhoxing-ms** [comment]
If the returned status is modified by CLI side, I am concerned that the experience of multiple client tools and REST is inconsistent. And if the PowerShell command `Get-AzProximityPlacementGroup` needs to return the colocation status of VMs in the future, the same logic needs to be added on PowerShell side. + PM @dcaro help take a look and give us some suggestions

**Comment by dcaro** [comment]
Azure cli and Azure PowerShell should be aligned on the response returned to the command line.
@Drewm3 / @jaylabell do you know why the `Get-AzProximityPlacementGroup` cmdlet interprets the `null` response from the API to return `Aligned`?

**Comment by ShivapriyaGujjarlapudi** [comment]
@dcaro Currently Get-AzProximityPlacementGroup cmdlet doesn't return colocation status of VMs.
To get colocation status of VMs, Get-AzVM cmdlet is used which is working fine, returning Aligned status.

**Comment by zhoxing-ms** [comment]
@dcaro In fact, both `Get-AzVM` in PowerShell and `az vm show` in CLI can get the colocation status of VM normally.
But at present, the `az ppg show` command in CLI directly outputs the colocation status of VMs as `null` which is returned by the REST service, while the `Get-AzProximityPlacementGroup` command in PowerShell does not display the colocation status.

Do you think CLI should change the output value of colocation status of VMs from `null` to `Aligned`, and then if `Get-AzProximityPlacementGroup` needs to display the colocation status in the future, the PowerShell command should also add the similar logic?
Or should the REST service fix the problem centrally and uniformly to ensure that the experience of multiple client tools and REST service is consistent?

**Comment by ashutoshtandon** [comment]
It was a conscious decision taken by design team for 'Aligned' status to make output less bloated. Aligned PPG implicitly means all underlying VMs are aligned.

**Comment by navba-MSFT** [comment]
Removing cxp attention label from this.

**Comment by Drewm3** [comment]
We are not going to change the REST API which would be a breaking change. I am OK if the CLI would like to drop the functionality to have it align with PowerShell.

**Comment by ShivapriyaGujjarlapudi** [comment]
Aligned with @Drewm3. This is a specific scenario where the PPG colocation status is "Aligned" where it implicitly means all underlying resources are aligned. The output for "Unknown" or "Not Aligned" can remain as it works today on CLI. Poral team would also be picking up these values accordingly.

Adding VM colocation status to PowerShell output in the future should also be ideally following these scenario rules.
